pkg/storaged/filesystem/utils.jsx

```diff
@@ -35,21 +35,65 @@ import { mounting_dialog } from "./mounting-dialog.jsx";
 
 const _ = cockpit.gettext;
 
-export function is_mounted(client, block) {
+/* is mounted with an entry in fstab */
+export function is_mounted(client, block, subvol) {
     const block_fsys = client.blocks_fsys[block.path];
     const mounted_at = block_fsys ? block_fsys.MountPoints : [];
-    const config = block.Configuration.find(c => c[0] == "fstab");
-    if (config && config[1].dir.v) {
-        let dir = decode_filename(config[1].dir.v);
-        if (dir[0] != "/")
-            dir = "/" + dir;
+    const [, dir] = get_fstab_config(block, false, subvol);
+    if (dir) {
         return mounted_at.map(decode_filename).indexOf(dir) >= 0;
     } else
         return null;
 }
 
-export function get_fstab_config(block, also_child_config) {
-    return get_fstab_config_with_client(client, block, also_child_config);
+export function get_fstab_config(block, also_child_config, subvol) {
+    return get_fstab_config_with_client(client, block, also_child_config, subvol);
+}
+
+/**
+ * Find all subvolumes with the given mountpoint in fstab without using
+ * get_fstab_config_with_client as it doesn't return multiple entries given the
+ * same block device.
+ */
+function find_subvols_for_mount_point(client, mount_point, subvol) {
+    const subvols = [];
+    const match = c => {
+        let self = false;
+
+        if (c[0] != "fstab")
+            return false;
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19690-20240108-153441-e8766e70-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)